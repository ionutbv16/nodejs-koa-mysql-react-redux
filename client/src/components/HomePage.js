import React from 'react';
import {callApiUpdate} from '../api/apiCall';

class HomePage extends React.Component {


  componentDidMount = () => {
    // UPDATE optimization table 
    callApiUpdate().then(response => {
        return  { response: response }; 
    });
  }


  render() {
    return (
      <div>
        <h1>Home Page</h1>
        <p>Welcome to MYSQL NodeJS React Redux Application !</p>
        <div className="ui message">
          <p>You can read more about features in Features/Specs section.</p>
        </div>
        <p className="ui header">FrontEnd Design:</p>
        <p>React.JS + Redux with the following dependencies</p>
        <p>'react' JavaScript library for building user interfaces</p>
        <p>'react-select' JavaScript library</p>
        <p>'react-dom' This package serves as the entry point of the DOM-related rendering paths</p>
        <p>'redux' Predictable state container for JavaScript apps</p>
        <p>'redux-devtools-extension' Redux DevTools extension</p>
        <p>'react-redux' Official React bindings for Redux</p>
        <p>'react-router-dom' Routing for React</p>
        <p>'semantic-ui-react' Semantic-UI-React is the official React integration for Semantic UI </p>
        <p>'classnames' A simple javascript utility for conditionally joining classNames together</p>
      </div>
    );
  }
}

export default HomePage ;
