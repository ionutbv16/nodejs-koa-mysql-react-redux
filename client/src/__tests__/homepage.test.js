import { shallow } from 'enzyme';

import sinon from 'sinon';
import HomePage from '../HomePage';


describe('<HomePage />', () => {
  it('renders three <HomePage /> components', () => {
    const wrapper = shallow(<HomePage />);
    
    //expect(wrapper.contains(<h1>Home Page</h1>)).to.equal(true); -- false 
    expect(wrapper).toMatchSnapshot();
  });
 
});