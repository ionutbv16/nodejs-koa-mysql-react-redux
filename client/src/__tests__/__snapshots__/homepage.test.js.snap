// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> renders three <HomePage /> components 1`] = `
<div>
  <h1>
    Home Page
  </h1>
  <p>
    Welcome to Events CRUD Application !
  </p>
  <p>
    This is a Create Read Update Delete Application for data imported from Facebook SDK API.
  </p>
  <div
    className="ui message"
  >
    <p>
      You can read more about features in   Features/Specs  section.
    </p>
  </div>
  <p
    className="ui header"
  >
    FrontEnd Design:
  </p>
  <p>
    React.JS + Redux with the following dependencies
  </p>
  <p>
    'react' JavaScript library for building user interfaces
  </p>
  <p>
    'react-dom' This package serves as the entry point of the DOM-related rendering paths
  </p>
  <p>
    'redux' Predictable state container for JavaScript apps
  </p>
  <p>
    'redux-thunk' Thunk middleware for Redux
  </p>
  <p>
    'redux-devtools-extension' Redux DevTools extension
  </p>
  <p>
    'react-redux' Official React bindings for Redux
  </p>
  <p>
    'react-router-dom' Routing for React
  </p>
  <p>
    'semantic-ui-react' Semantic-UI-React is the official React integration for Semantic UI 
  </p>
  <p>
    'classnames' A simple javascript utility for conditionally joining classNames together
  </p>
  <p
    className="ui header"
  >
    BackEnd Design:
  </p>
  <p>
    Node.JS + MongoDB with the following dependencies
  </p>
  <p>
    'express' Node.js web application framework
  </p>
  <p>
    'body-parser' Node.js body parsing middleware
  </p>
  <p>
    'mongoose' Elegant MongoDB object modeling for Node.js
  </p>
  <p>
    'tryjson' Faster try/catch helper for JSON.parse
  </p>
  <p>
    'cors' enable cross-origin resource sharing
  </p>
</div>
`;
